### DISCUSSION TITLE
Support nameof() in CallerArgumentExpression

### DISCUSSION THREAD

**Discussion by CommonLoon102** [dotnet/roslyn#57792] [Ideas] [answered]
Without `nameof()`, it is not possible to rename parameter names safely. Currently, if I try to use nameof(), it says that "The name 'a' does not exist in the context."

https://weblogs.asp.net/dixin/csharp-10-new-feature-callerargumentexpression-argument-check-and-more

```
void Function(int a, TimeSpan b, [CallerArgumentExpression(nameof(a))] string c = "", [CallerArgumentExpression("b")] string d = "")
{
    Console.WriteLine($"Called with value {a} from expression '{c}'");
    Console.WriteLine($"Called with value {b} from expression '{d}'");
}
```

**Comment by alrz** [comment, marked as answer]
This is championed at https://github.com/dotnet/csharplang/issues/373
